src/Discovery/src/EurekaBase/EurekaInstanceOptions.cs

```diff
@@ -162,5 +136,54 @@ public override string GetHostName(bool refresh)
 
             return _thisHostName;
         }
+
+        public void ApplyConfigUrls(List<Uri> addresses, string wildcard_hostname)
+        {
+            // only use addresses from config if there are any and registration method hasn't been set
+            // if registration method has been set, the user probably wants to define their own behavior
+            if (addresses.Any() && string.IsNullOrEmpty(RegistrationMethod))
+            {
+                foreach (var address in addresses)
+                {
+                    if (address.Scheme == "http" && Port == DEFAULT_NONSECUREPORT)
+                    {
+                        Port = address.Port;
+                    }
+                    else if (address.Scheme == "https" && SecurePort == DEFAULT_SECUREPORT)
+                    {
+                        SecurePort = address.Port;
+                    }
+
+                    if (!address.Host.Equals(wildcard_hostname) && !address.Host.Equals("0.0.0.0"))
+                    {
+                        HostName = address.Host;
+                    }
+                }
+            }
+        }
+
+        public void SetInstanceId(IConfiguration config)
+        {
+            var defaultId = GetHostName(false) + ":" + EurekaInstanceOptions.Default_Appname + ":" + EurekaInstanceOptions.Default_NonSecurePort;
+            if (defaultId.Equals(InstanceId))
+            {
+                var springInstanceId = config.GetValue<string>(EurekaPostConfigurer.SPRING_APPLICATION_INSTANCEID_KEY);
+                if (!string.IsNullOrEmpty(springInstanceId))
+                {
+                    InstanceId = springInstanceId;
+                }
+                else
+                {
+                    if (SecurePortEnabled)
+                    {
+                        InstanceId = GetHostName(false) + ":" + AppName + ":" + SecurePort;
```

Comment:
@hananiel this line actually _should_ be hit by the same test that hits the block under line 152 above... should we set SecurePortEnabled=true when using urls from config that use https? Consul only supports a single port and this PR sets prefers https, so maybe we should add something like that here, what do you think?